src/middlewares/hooks/index.ts

```diff
@@ -296,9 +303,16 @@ export class HooksManager {
 
     if (hook.type === 'guardrail' && hook.checks) {
       const checkResults = await Promise.all(
-        hook.checks.map((check: Check) =>
-          this.executeFunction(span.getContext(), check, hook.eventType)
-        )
+        hook.checks
+          .filter((check: Check) => check.is_enabled !== false)
```

Comment:
`is_enabled` is an optional field which maybe `undefined` and that can cause this condition true. Can you check?!

Reply:
As per the ts interface for `Check` `is_enabled` can be `undefined`. Now `undefined !== false` is resulting `true`.

If we're sure that `is_enabled` is always available,  we can remove optional specifier in type def.

Reply:
is_enabled is not a mandatory field. If its not set, we assume the default behaviour which is true. Until its explicitly set as false, we will consider it as valid